streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -864,6 +941,52 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
         }
     }
 
+  /**
+   * More powerful version of `ZStream#groupByKey`
+   */
+  def groupBy[R1 <: R, E1 >: E, K, V](
+    f: A => ZIO[R1, E1, (K, V)],
+    buffer: Int = 12
```

Comment:
16 is probably better than 12 since the underlying `RingBuffer` has better performance with power of 2s

Reply:
Ah nice. Updated